### PR TITLE
Add the project documentation

### PR DESCRIPTION
**Pull request by mprahl**
As part of documenting the coding standards, I introduced the `flake8-docstrings` plugin which enforces our use of docstrings and the proper format. There were a few non-compliant docstrings that were fixed in this PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ Get the IIB build request from the REST API.
 :param int request_id: the ID of the IIB request
 :return: the request
 :rtype: dict
-:raises iib.exceptions.IIBError: if the HTTP request fails
+:raises IIBErrorequest fails
```
Comment (yashvardhannanavati) [inline README.md]: should this be `:raises IIBError: if the HTTP request fails`

File: README.md [hunk 2]
```diff
@@ -1,6 +1,38 @@
...
+```
+
+If you are using Docker (a modern version is required), you can perform the following:
+```bash
+sudo docker login -u iib -p iibpassword localhost:8443
+sudo docker run localhost:8443/iib-build:1
+```
+
+If your development environment requires accessing a private container registry, please read
+the section titled Registry Authentication.
```
Comment (yashvardhannanavati) [inline README.md]: Can we have a hyperlink to the `Registry Authentication` section?

File: README.md [hunk 3]
```diff
@@ -147,3 +147,34 @@ The custom configuration options for the Celery workers are listed below:
...
+
+To build and serve the docs, run the following commands:
+```
+$ tox -e docs
+$ google-chrome .tox/docs_out/index.html
+```
+
+### Expanding the Docs
+
+To document a new Python module, find the `rst` file of the Python package that contains the module.
```
Comment (yashvardhannanavati) [inline README.md]: Optional: `find the `rst` file of the Python package that contains the module` => `find the `rst` file of the corresponding Python package that contains the module`

File: README.md (hunk 2 above)
Comment (mprahl) [reply README.md]: I can have it work on GitHub or the Read the Docs but not both. I'll skip this suggestion.

**Comment by mprahl** [comment]
@yashvardhannanavati could you please take a look again?

**Pull request merged**